### ISSUE TITLE
Feauture request: ISM policy for OpenSearch

### ISSUE THREAD

**Issue by brainfair** [xco-sk/eck-custom-resources#60]
Unfortunately in OpenSearch, it is forbidden to create ILM policies. Instead, there is an ISM policy that we can create and attach.
That would be nice also to manage OpenSearch instances via eck-custom-resources.

https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ism.html

**Comment by xco-sk** [comment]
Hi @brainfair, sorry for the delayed response.

TL;DR answer: We might add support for ISM in the future, but maybe not in this project, but rather in the OpenSearch fork, with other OpenSearch relevant resources support. 

Longer answer:
The OpenSearch is "officially" not supported, even though some resources work with it just fine because of its similarity with API. The original idea behind this operator was to add a missing support for _other_ resources, that are not supported via Elastic's official ECK operator - that's why it's even part of the project name. Throughout the time, we removed this coupling to ECK, and this operator can now be used with any kind of Elasticsearch deployment. That's for the history, now the OpenSearch. The issue is, that there are multiple resources that do not work with OS correctly, namely the ILMs, Users, Roles, and maybe others too. It would make more sense to fork this operator and create an OS flavor, with full-blown support, that is unfortunately not on the roadmap at the moment, considering the limited time we have for this project.
Sorry for a bit disappointing news.

**Comment by brainfair** [comment]
Hi! Thanks for the information, I think it can be closed then, maybe in the future if I have time I will make improvements myself.
